tests/test_pipelines_sentiment_analysis.py

```diff
@@ -1,12 +1,90 @@
 import unittest
 
+import numpy as np
+
+from transformers import AutoTokenizer, DistilBertConfig, DistilBertForSequenceClassification, pipeline
+from transformers.testing_utils import slow
+
 from .test_pipelines_common import MonoInputPipelineCommonMixin
 
 
+VALID_INPUTS = ["I really disagree with what you've said.", ["I love you."]]
+
+
 class SentimentAnalysisPipelineTests(MonoInputPipelineCommonMixin, unittest.TestCase):
     pipeline_task = "sentiment-analysis"
-    small_models = [
-        "sshleifer/tiny-distilbert-base-uncased-finetuned-sst-2-english"
-    ]  # Default model - Models tested without the @slow decorator
+    small_models = ["distilbert-base-cased"]  # Default model - Models tested without the @slow decorator
     large_models = [None]  # Models tested with the @slow decorator
     mandatory_keys = {"label", "score"}  # Keys which should be in the output
+
+    @slow
+    def test_function_to_apply(self):
+        for model_name in self.small_models:
+            string_input, string_list_input = VALID_INPUTS
+            tokenizer = AutoTokenizer.from_pretrained(model_name, use_fast=True)
+
+            model = DistilBertForSequenceClassification(DistilBertConfig.from_pretrained(model_name))
+            model.eval()
+            classifier = pipeline(task="sentiment-analysis", model=model, tokenizer=tokenizer)
+
+            def pipeline_call_argument(argument=None):
+                _string_output = classifier(string_input, function_to_apply=argument)
+                _string_list_output = classifier(string_list_input, function_to_apply=argument)
+                return _string_output, _string_list_output
+
+            def pipeline_init_argument(argument=None):
+                _classifier = pipeline(
+                    task="sentiment-analysis", model=model, tokenizer=tokenizer, function_to_apply=argument
+                )
+                _string_output = _classifier(string_input)
+                _string_list_output = _classifier(string_list_input)
+                return _string_output, _string_list_output
+
+            def pipeline_model_argument(argument=None):
+                model.config.task_specific_params = {"sentiment-analysis": {"function_to_apply": argument}}
+                _classifier = pipeline(task="sentiment-analysis", model=model, tokenizer=tokenizer)
+                _string_output = _classifier(string_input)
+                _string_list_output = _classifier(string_list_input)
+                return _string_output, _string_list_output
+
+            string_output = classifier(string_input)
+            string_list_output = classifier(string_list_input)
```

Comment:
I think we're missing from these tests an actual tests of what `string_output` and `string_list_output` look like:

Ideally `self.assertEqual(string_output, {'label': XX, 'score': XX})`.
Realistically because assertEqual won't work on floats:

```python
assertTrue(isinstance(string_output, dict))
assertTrue(set(string_output.keys()), {'label', 'score'})
assertEquals(string_output['score'].shape, [12, 1064])
# checking only one value is probably good enough, but at least we will know if the output changes
assertAlmostEquals(string_output['score'][0, 0], -12.125)
```